tools/generators/target_build_settings/src/Generator/ParseTransitiveSwiftDebugSettings.swift

```diff
@@ -0,0 +1,145 @@
+                )
+            }
+            guard let clangArgsCount = Int(rawClangArgsCount) else {
+                throw PreconditionError(message: url.prefixMessage("""
+Clang args count "\(rawClangArgsCount)" was not an integer
+"""))
+            }
+            argsLoop: for index in (0..<clangArgsCount) {
+                guard let arg =
+                    try await iterator.next()?.nullsToNewlines
+                else {
+                    throw PreconditionError(message: url.prefixMessage("""
+Too clang args. Found \(index), expected \(clangArgsCount)
+"""))
+                }
+
+                for onceArgPrefixes in clangOnceArgPrefixes {
+                    if arg.hasPrefix(onceArgPrefixes) {
+                        guard !onceClangArgs.contains(arg) else {
+                            continue argsLoop
+                        }
+                        onceClangArgs.insert(arg)
+                        break
+                    }
+                }
+                clangArgs.append(arg)
+            }
+
+            guard let rawFrameworkIncludesCount = try await iterator.next()
+            else {
+                throw PreconditionError(
+                    message:
+                        url.prefixMessage("Missing framework includes count")
+                )
+            }
+            guard let frameworkIncludesCount = Int(rawFrameworkIncludesCount)
+            else {
+                throw PreconditionError(message: url.prefixMessage("""
+Framework includes count "\(rawFrameworkIncludesCount)" was not an integer
+"""))
+            }
+            for index in (0..<frameworkIncludesCount) {
+                guard let include =
+                    try await iterator.next()?.nullsToNewlines
+                else {
+                    throw PreconditionError(message: url.prefixMessage("""
+Too few framework includes. Found \(index), expected \(frameworkIncludesCount)
+"""))
+                }
+                frameworkIncludes.append(include)
+            }
+
+            guard let rawSwiftIncludesCount = try await iterator.next()
+            else {
+                throw PreconditionError(
+                    message: url.prefixMessage("Missing swift includes count")
+                )
+            }
+            guard let swiftIncludesCount = Int(rawSwiftIncludesCount)
+            else {
+                throw PreconditionError(message: url.prefixMessage("""
+Swift includes count "\(rawSwiftIncludesCount)" was not an integer
+"""))
+            }
+            for index in (0..<swiftIncludesCount) {
+                guard let include =
+                    try await iterator.next()?.nullsToNewlines
+                else {
+                    throw PreconditionError(message: url.prefixMessage("""
+Too few swift includes. Found \(index), expected \(swiftIncludesCount)
+"""))
+                }
+                swiftIncludes.append(include)
```

Comment:
Nit: this pattern of "casting to `Int` => iterating => checking if 'too few' iterations => `.append`-ing" repeats a few times here. Any reason to not extract this to a helper func?

Reply:
Fixed in #2708.